test/test_hpyerr.py

```diff
@@ -461,14 +461,20 @@ def test_HPyErr_WarnEx(self):
         """)
 
         # NOTE: trampoline is defined in support.py
+        import sys
+        ISPYPY = "__pypy__" in sys.modules
         def check_warning(arg, category, message, file):
             with warnings.catch_warnings(record=True) as warnings_list:
-                trampoline(mod.f, arg)
+                if ISPYPY:
+                    mod.f(arg)
+                else:
+                    trampoline(mod.f, arg)
```

Comment:
Did the original author mean to call the legacy function here?

Reply:
I have no idea why `trampoline` was invoked here. Perhaps we can ask in the HPy Discord.

Reply:
I think we need to call `trampoline`, such that the `support.py` file ends up on the call-stack when the warning is signaled. This helper method is [called like this](https://github.com/hpyproject/hpy/pull/455/files/f90e45bd2b65a8aac2e646b84a00e0e1f8112459#diff-193925ea633541353994533e8c9fa601c71e47456b6210a3b451aa5ec9622f75R479):
```
check_warning(0, RuntimeWarning, "warn qzp", "support.py")
```
and the assert that you disabled for PyPy should actually check that "support.py" in this case is as a filename in the warning object:
```
assert w.filename.endswith(file) # where file will be `support.py` for us
```